gnss_lib_py/parsers/precise_ephemerides.py

```diff
@@ -0,0 +1,811 @@
+
+    Notes
+    -----
+    The format for .sp3 files can be viewed in [1]_.
+
+    This parser function does not process all available GNSS constellations
+    at once, i.e., needs to be independently called for each desired one
+
+    0th array of the Clk class is always empty since PRN=0 does not exist
+
+    Based on code written by J. Makela.
+    AE 456, Global Navigation Sat Systems, University of Illinois
+    Urbana-Champaign. Fall 2015
+
+    References
+    ----------
+    .. [1]  https://files.igs.org/pub/data/format/sp3d.pdf
+            Accessed as of August 20, 2022
+    """
+    # Initial checks for loading sp3_path
+    if not isinstance(input_path, str):
+        raise TypeError("input_path must be string")
+    if not os.path.exists(input_path):
+        raise OSError("file not found")
+
+    # Load in the file
+    with open(input_path, 'r', encoding="utf-8") as infile:
+        data = [line.strip() for line in infile]
+
+    # Poll the total no. of satellites based on constellation specified
+    if constellation in NUMSATS.keys():
+        nsvs = NUMSATS[constellation][0]
+    else:
+        raise RuntimeError("No support exists for specified constellation")
+
+    # Create a sp3 class for each expected satellite
+    sp3data = []
+    for _ in np.arange(0, nsvs+1):
+        sp3data.append(Sp3())
+        sp3data[-1].const = constellation
+
+    # Loop through each line
+    for dval in data:
+        if len(dval) == 0:
+            # No data
+            continue
+
+        if dval[0] == '*':
+            # A new record
+            # Get the date
+            temp = dval.split()
+            curr_time = datetime( int(temp[1]), int(temp[2]), \
+                                  int(temp[3]), int(temp[4]), \
+                                  int(temp[5]),int(float(temp[6])), \
+                                  tzinfo=timezone.utc )
+            gps_millis = datetime_to_gps_millis(curr_time, add_leap_secs = False)
+
+        if 'P' in dval[0]:
+            # A satellite record.  Get the satellite number, and coordinate (X,Y,Z) info
+            temp = dval.split()
+
+            if temp[0][1] == NUMSATS[constellation][1]:
+                prn = int(temp[0][2:])
+                sp3data[prn].utc_time.append(curr_time)
+                sp3data[prn].tym.append(gps_millis)
+                sp3data[prn].xpos.append(float(temp[1])*1e3)
+                sp3data[prn].ypos.append(float(temp[2])*1e3)
+                sp3data[prn].zpos.append(float(temp[3])*1e3)
+
+    # Add warning in case any satellite PRN does not have data
+    no_data_arrays = []
+    for prn in np.arange(1, nsvs+1):
+        if len(sp3data[prn].tym) == 0:
+            no_data_arrays.append(prn)
+    if len(no_data_arrays) == nsvs:
+        warnings.warn("No sp3 data found for PRNs: "+str(no_data_arrays), RuntimeWarning)
+
+    return sp3data
+
+class Clk:
+    """Class handling satellite clock bias data from precise ephemerides
+
```

Comment:
Can you add the parameters of the class in the docstring for the class? For both this one and the SP3 class